client/components/site-selector-modal/index.jsx

```diff
@@ -98,11 +85,28 @@ class SiteSelectorModal extends Component {
 				</div>
 				<SitesDropdown
 					onSiteSelect={ this.setSite }
-					selectedSiteId={ this.state.site && this.state.site.ID }
+					selectedSiteId={ this.state.siteId }
 					filter={ this.props.filter } />
 			</Dialog>
 		);
 	}
 }
 
-export default localize( SiteSelectorModal );
+export default connect(
+	( state, { filter } ) => {
+		const primarySiteId = getPrimarySiteId( state );
+		const visibleSites = getVisibleSites( state );
+
+		let filteredSiteIds = visibleSites.map( ( site ) => site.ID );
+
+		if ( filter ) {
+			filteredSiteIds = filteredSiteIds.filter( filter );
+		}
+
+		return {
+			initialSiteId: includes( filteredSiteIds, primarySiteId )
+				? primarySiteId
+				: filteredSiteIds[ 0 ]
```

Comment:
I... hope not? 😄 We've had this part of the logic [before](https://github.com/Automattic/wp-calypso/pull/13293/files#diff-f7f3055cc9ffe5ea8a7a5a5712368a8eL54), so I'm not too worried, TBH.